ts/Extensions/Breadcrumbs.ts

```diff
@@ -0,0 +1,1103 @@
+            breadcrumbs.redraw();
+        }
+    });
+
+    addEvent(Chart, 'redraw', function (): void {
+        this.breadcrumbs && this.breadcrumbs.redraw();
+    });
+
+    addEvent(Chart, 'beforeDrillUp', function (): void {
+        if (this.breadcrumbs) {
+            this.breadcrumbs.isDirty = true;
+            this.breadcrumbs.resetList();
+        }
+    });
+
+    addEvent(Chart, 'applyDrilldown', function (): void {
+        const chart = this,
+            breadcrumbs = chart.breadcrumbs as Highcharts.Breadcrumbs,
+            breadcrumbsOptions = chart.options.drilldown && chart.options.drilldown.breadcrumbs;
+
+        if (!breadcrumbs) {
+            chart.breadcrumbs = new Breadcrumbs(chart as Chart, breadcrumbsOptions, false);
+            chart.breadcrumbs.createList();
+
+        } else {
+            breadcrumbs.isDirty = true;
+            breadcrumbs.redraw();
+        }
+
+    });
+
+    let treemapEventsAdded: boolean = false;
+
+    // Causes some troubles in karma tests.
+    addEvent(Chart, 'init', function (): void {
+        if (H.seriesTypes.treemap && !treemapEventsAdded) {
+
+            addEvent(H.seriesTypes.treemap,
+                'setRootNode',
+                function (this: TreemapSeries, e: TreemapSeries.SetRootNodeObject): void {
+                    const chart = this.chart,
+                        breadcrumbsOptions = this.options.breadcrumbs;
+
+                    if (!chart.breadcrumbs) {
+                        chart.breadcrumbs = new Breadcrumbs(chart as Chart, breadcrumbsOptions, true);
+                    }
+                    if (chart.breadcrumbs) {
+                        chart.breadcrumbs.isDirty = true;
+                        // Create a list using the event after drilldown.
+                        if ((e as any).trigger === 'click' || !(e as any).trigger) {
+                            chart.breadcrumbs.createList(e);
+                            chart.breadcrumbs.draw();
+                        }
+
+                        if ((e as any).trigger === 'traverseUpButton') {
+                            chart.breadcrumbs.resetList();
+                        }
+                    }
+                });
+            addEvent(H.seriesTypes.treemap, 'update', function (e: any): void {
+                const breadcrumbs = this.chart.breadcrumbs;
+
+                if (breadcrumbs && e.options.breadcrumbs) {
+                    breadcrumbs.isDirty = true;
+                    breadcrumbs.update(e.options.breadcrumbs as Highcharts.BreadcrumbsOptions, false);
+                    breadcrumbs.redraw();
+                }
+            });
+            treemapEventsAdded = true;
+        }
+    });
+
+    // Remove resize/afterSetExtremes at chart destroy
+    addEvent(Chart, 'destroy', function destroyEvents(): void {
+        if (this.breadcrumbs) {
+            this.breadcrumbs.destroyGroup();
+            this.breadcrumbs = void 0 as any;
```

Comment:
Why `as any`? Shouldn't Chart.breadcrumbs be `Breadcrumbs|undefined`?

Reply:
I would like to set `this.breadcrumbs = undefined`, is there any other way to do this without using `void 0 as any`?

Reply:
Chart.breadcrumbs is optional in ChartLike, but does not allow `undefined` as a possible type. Extend the type there like Torstein wrote, or use here `delete chart.breadcrumbs` instead of a `void 0` assignment.